plugins/inputs/http_listener_v2/http_listener_v2.go

```diff
@@ -158,16 +166,37 @@ func (h *HTTPListenerV2) Start(acc telegraf.Accumulator) error {
 
 	h.acc = acc
 
-	server := h.createHTTPServer()
+	h.server = h.createHTTPServer()
+
+	go func() {
+		// This goroutine just moves data from the metrics channel to the accumulator
+		// It exists so we can select on writing to the metrics channel and handle cancellation correctly
+		// in the request handler.
+		defer func() {
+			// Handle the case where the accumulator channel is closed while shutting down. We do this to
+			// avoid changing the accumulator API to allow for more graceful handling of this channel.
+			if err := recover(); err != nil {
+				h.Log.Error(err)
+			}
+		}()
+		for {
+			metrics, ok := <-h.metrics
```

Comment:
Perhaps add some conditions around this channel to prevent this goroutine hanging forever on a `nil` channel.

Reply:
We manage this channel ourselves, it can't ever be nil.

Reply:
Got it but think about e.g. tests? `init` might not be called then

Reply:
But OTOH below `if` already checks that 👍